dockeradapter/dockeradapter.go

```diff
@@ -130,18 +141,91 @@ func ImageExists(ctx context.Context, client ImageInspector, imageName string) b
 	return true
 }
 
-// New returns a client satisfying the Client interface
+// New returns a client connected with the highest API version supported by both the Lumogon client
+// and the Docker runtime
 func New() (Client, error) {
-	concreteClient := new(concreteDockerClient)
-	logging.Debug("[Docker Adapter] Creating container runtime client: Docker")
-	dockerAPIClient, err := client.NewEnvClient()
+
+	host, _, _ := DockerEnvvars()
+
+	serverAPIVersion, _, err := ServerInfo(host)
 	if err != nil {
-		return nil, fmt.Errorf("[Docker Adapter] Unable to initialise container runtime type: Docker, error: %s", err)
+		logging.Debug("[Docker Adapter] Unable to determine Docker Server API version: %s", err.Error())
+		return nil, err
 	}
-	concreteClient.Client = dockerAPIClient
+
+	// Connect using the API version that the server supports
+	client, err := client.NewClient(host, serverAPIVersion, nil, nil)
+	if err != nil {
+		logging.Debug("[Docker Adapter] Unable to connect to Docker server")
+		return nil, err
+	}
+
+	logging.Debug("[Docker Adapter] Creating container runtime client: Docker")
+	concreteClient := new(concreteDockerClient)
+	concreteClient.APIVersion = serverAPIVersion
+	concreteClient.Client = client
+
 	return concreteClient, nil
 }
 
+// DockerEnvvars returns values for the following environment variables, setting a default
+// if no variable is set: DOCKER_HOST, DOCKER_CERT_PATH, DOCKER_TLS_VERIFY
+func DockerEnvvars() (string, string, bool) {
+	logging.Debug("[Docker Adapter] Negotiating client connection with Docker server")
+	host, ok := os.LookupEnv("DOCKER_HOST")
+	if !ok {
+		host = "unix:///var/run/docker.sock"
+	}
+	logging.Debug("[Docker Adapter] Setting Docker host to: %s", host)
+
+	certPath, ok := os.LookupEnv("DOCKER_CERT_PATH")
+	if !ok {
+		certPath = ""
+	}
+	logging.Debug("[Docker Adapter] Setting cert path to: %s", certPath)
+
+	var tlsVerify bool
+	s, ok := os.LookupEnv("DOCKER_TLS_VERIFY")
+	if !ok {
+		tlsVerify = false
+	} else {
+		var parseBoolErr error
+		tlsVerify, parseBoolErr = strconv.ParseBool(s)
+		if parseBoolErr != nil {
+			logging.Debug("Error parsing DOCKER_TLS_VERIFY envvar, setting to false: %s", parseBoolErr.Error())
```

Comment:
I feel this should at least be a higher-level always visible message, or even an error. ie. If I incorrectly pass something here then my security is degraded but finding that out is hard. Better to fail than be helpful here I think.

Reply:
The TLS envvars are placeholder at the moment, its not actually used right now. Should probably remove until that gets added.